docs: component updates 2025-41 DOC-2300 (#8337)
spectrocloud/librarium: docs/docs-content/release-notes/release-notes.md

<ReleaseNotesVersions />

## October 10, 2025 - Component Updates {#component-updates-2025-41}

The following components have been updated for Palette version 4.7.20 - 4.7.23.

| Component                                                                                                         | Version |
| ----------------------------------------------------------------------------------------------------------------- | ------- |
| [Spectro Cloud Terraform provider](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs) | 0.25.0  |
| [Spectro Cloud Crossplane provider](https://marketplace.upbound.io/providers/crossplane-contrib/provider-palette) | 0.25.0  |

### Breaking Changes

- To avoid unnecessary cluster repaves, the
  [Spectro Cloud Terraform provider](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs) no
  longer fails if a timeout occurs during cluster provisioning. Instead, the provider logs an error and continues
  provisioning in the background. Future Terraform executions reconcile the state of the cluster with the Terraform
  state.

>   The
>   [`spectrocloud_cluster` data source](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/data-sources/cluster)
>   now exposes state and health attributes. We recommend using these to check that you can use to validate cluster
>   readiness before triggering any [cluster management](../clusters/cluster-management/cluster-management.md) operations.

Reviewer: ```suggestion
  
  For increased transparency in these situations, the
  [`spectrocloud_cluster` data source](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/data-sources/cluster)
  now exposes state and health attributes. We recommend using these to validate cluster
  readiness before triggering any [cluster management](../clusters/cluster-management/cluster-management.md) operations.
```

Reviewer: "We recommend using these" - Can we clarify what "these" are? The attributes? Issuing a command?

Author: the new attributes. will clarify

### Improvements

- The

>   [`spectrocloud_virtual_machine` resource](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine)

Reviewer: ```suggestion
  [`spectrocloud_virtual_machine` Terraform resource](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine)
```

>   has the following improvements:

Reviewer: ```suggestion
  now supports the following configurations:
```

>   - The resource now supports the `network_data` configuration under the
>     [`cloud_init_no_cloud` set](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#cloud_init_no_cloud-1),
>     allowing you to supply network configuration when provisioning VMs.
>   - The resource now supports DataVolume storage specification under the
>     [`data_volume_templates.spec` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-data_volume_templatesspec).
>   - The resource now supports VM creation with blank `pvc` and `storage` fields under the
>     [`data_volume_templates.spec` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-data_volume_templatesspec),
>     allowing you to create VMs without these specifications.
>   - The resource now supports `boot_order` configuration under the
>     [`disk` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-disk),
>     allowing you to specify the ordering of boot devices.

Reviewer: ```suggestion
  - `network_data` under the
    [`cloud_init_no_cloud` set](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#cloud_init_no_cloud-1),
    allowing you to supply network configurations when provisioning VMs.
  - DataVolume storage under the
    [`data_volume_templates.spec` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-data_volume_templatesspec).
  - VM creation with blank `pvc` and `storage` fields under the
    [`data_volume_templates.spec` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-data_volume_templatesspec),
    allowing you to create VMs without these specifications.
  - `boot_order` under the
    [`disk` list](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/virtual_machine#nested-schema-for-disk),
    allowing you to specify the order of boot devices.
```

### Bug Fixes

> - Fixed an issue that caused repeated reconciles when a
>   [`spectrocloud_backup_storage_location` resource](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/backup_storage_location)
>   was specified. This was due to the AWS secret key not being marked as sensitive.

Reviewer: ```suggestion
- Fixed an issue that caused repeated reconciliation when specifying a
  [`spectrocloud_backup_storage_location` Terraform resource](https://registry.terraform.io/providers/spectrocloud/spectrocloud/latest/docs/resources/backup_storage_location) due to AWS secret keys not being marked as sensitive.
```
